utils/githooks/pre-commit.d/20-flake.sh

```diff
@@ -22,6 +22,11 @@ if ! command -v flake8 > /dev/null 2>&1; then
     exit 0
 fi
 
+if flake8 --version | grep ^6\\.; then
+    echo "  Flake8 >= 6.x does not have the --diff option, skipping."
+    exit 0
+fi
+
```

Comment:
I'm working on a more proper solution for "package is installed but the wrong version". And this check in particular should be updated to not rely on `--diff`.
But why not just install the correct version of flake8? Why have flake8 installed at all if it's the wrong version?
Even the pylint, yamllint, isort (maybe others) behave in unexpected ways if the wrong version is installed.

Reply:
> I'm working on a more proper solution for "package is installed but the wrong version". And this check in particular should be updated to not rely on --diff.

That's good.

> But why not just install the correct version of flake8? Why have flake8 installed at all if it's the wrong version?

Because flake is installed on my machine by the O/S (i.e. as an RPM) and it's newer than what our _cq/requirements.txt_ is holding us back at.  It is used by other tools such as VS Code, etc.

Reply:
Ah :( This is an example where we should be running in a virtual env. Even if we supported newer flake, there would be subtle differences between the version expected by the tool and the version installed by the OS.
Anyway, not a blocker here. Just trying to understand the problem